packages/amplify-category-function/src/provider-utils/awscloudformation/utils/packageLayer.ts

```diff
@@ -108,17 +111,54 @@ async function zipLayer(context: $TSContext, resource: PackageRequestMeta, hash:
   });
 }
 
-// Check hash results for content changes, bump version if so
-async function ensureLayerVersion(context: $TSContext, layerName: string, previousHash: string) {
-  const currentHash = await hashLayerVersionContents(getLayerPath(layerName));
-  const isNewVersion = previousHash !== currentHash;
-  if (isNewVersion) {
-    if (previousHash) {
-      context.print.success(`Content changes in Lambda layer ${layerName} detected.`);
-      context.print.warning('Note: You need to run "amplify update function" to configure your functions with the latest layer version.');
+export async function checkContentChanges(context: $TSContext, resources: Array<any>): Promise<void> {
+  const changedResources = await getChangedResources(resources);
+
+  if (changedResources.length > 0) {
+    context.print.info('Content changes in Lambda layers detected.');
+    context.print.info('Suggested configuration for new layer versions:');
+    context.print.info('');
+
+    const timestampString = new Date().toISOString();
+    const prepushNotificationMessage = changedResources.map(resource => {
+      const { resourceName } = resource;
+      return ` ${resourceName}
+      - ${accessPermissions}: ${chalk.green('Maintain existing permissions')}
+      - ${description}: ${chalk.green('Updated layer version ') + chalk.gray(timestampString)}`;
+    });
+    context.print.info(prepushNotificationMessage.join(EOL));
+    context.print.info('');
+
+    const accepted = await context.prompt.confirm('Accept the suggested layer version configurations?', true);
+    for (const resource of changedResources) {
+      let parameters = loadStoredLayerParameters(context, resource.resourceName);
+      if (accepted) {
+        parameters = await lambdaLayerNewVersionWalkthrough(parameters, timestampString);
+      } else {
+        parameters.description = `Updated layer version ${timestampString}`;
+      }
```

Comment:
This logic seems backwards to me. If the configuration is accepted, then shouldn't we skip the prompt?

Also, we need to make sure the --yes flag is respected here